atlasdb-config/src/main/java/com/palantir/atlasdb/factory/ConfigRefreshable.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory;
+
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.atlasdb.config.AtlasDbRuntimeConfig;
+import com.palantir.common.concurrent.PTExecutors;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import com.palantir.refreshable.DefaultRefreshable;
+import com.palantir.refreshable.Refreshable;
+
+final class ConfigRefreshable implements AutoCloseable {
+
+    private static final Logger log = LoggerFactory.getLogger(ConfigRefreshable.class);
+    private static final Duration REFRESH_INTERVAL = Duration.ofSeconds(1);
+
+    private final Refreshable<Optional<AtlasDbRuntimeConfig>> delegate;
+    private final Runnable closer;
+
+    private ConfigRefreshable(
+            Refreshable<Optional<AtlasDbRuntimeConfig>> delegate,
+            Runnable closer) {
+        this.delegate = delegate;
+        this.closer = closer;
+    }
+
+    public Refreshable<Optional<AtlasDbRuntimeConfig>> refreshable() {
+        return delegate;
+    }
+
+    @Override
+    public void close() {
+        closer.run();
+    }
+
+    static ConfigRefreshable wrap(Refreshable<Optional<AtlasDbRuntimeConfig>> delegate) {
```

Comment:
Seems kinda odd to wrap the refreshable in this way. I think it was expose a way to convert a supplier to an optional using some executor, and handle the lifecycle of the executor seperately

Reply:
The suggestion is to just get rid of the extra level of abstraction and manage the lifecycle of the executor used to update the refreshables separately. This is especially beneficial since it allows you to reuse the executor across refreshables and allows you to avoid unnecessarily wrapping existing refreshables.

Hence:
```
ConfigRefreshable<T> config1 = ConfigRefreshable.createPolling(mySupplier1);
ConfigRefreshable<T> config2 = ConfigRefreshable.createPolling(mySupplier2);
...

config1.close();
config2.close();
```

becomes:
```
ScheduledExecutorService refreshableExecutor = PTExecutors.newSingleThreadScheduledExecutor();

Refreshable<T> config1 = Refreshables.create(mySupplier1, refreshableExecutor);
Refreshable<T> config2 = Refreshables.create(mySupplier2, refreshableExecutor);
...

refreshableExecutor.shutdown()
```

Reply:
I don’t expect to be using refreshables anywhere else. Handling this separately in TransactionManagers will be messy, because you don’t create the executor if someone provides the Refreshable. I can see what it would look like though

Reply:
So I tried to come up with something that's not messy and I can't: the fact that you only sometimes create the executor, and then you need to add it to the closeables list AND THEN, but only after TransactionManager is created, call #shutdownCallbacks with it makes me not want to do it.

This is actually a nice small abstraction of a closeable refreshable, and everything falls through nicely.

Reply:
Instead, I don't expose any special wrap methods, but instead the only way to create this special refreshable is from the TransactionManagers builder.

Reply:
And if we excavate this so we can remove the deprecated method, we'll just remove all this code.

Reply:
sure, whatever you're happy with